scripts/ssl_tts/make_supdata.py

```diff
@@ -0,0 +1,434 @@
+    def _is_valid_pitch(pitch_mean, pitch_std):
+        c1 = pitch_mean > 0 and pitch_mean < 1000
+        c2 = pitch_std > 0 and pitch_std < 1000
+        return c1 and c2
+
+    sup_data_dir = SUP_DATA_DIR
+    speaker_wise_pitch_contours = {}
+    for item in wav_and_id_list:
+        _, wav_id, speaker = item
+        pitch_contour_fn = f"pitch_contour_{wav_id}.pt"
+        pitch_contour_fp = os.path.join(sup_data_dir, pitch_contour_fn)
+        if speaker not in speaker_wise_pitch_contours:
+            speaker_wise_pitch_contours[speaker] = []
+        speaker_wise_pitch_contours[speaker].append(pitch_contour_fp)
+
+    speaker_pitch_stats = {}
+    for speaker in speaker_wise_pitch_contours:
+        non_zero_pc = []
+        for pitch_contour_fp in speaker_wise_pitch_contours[speaker][:50]:
+            pitch_contour = torch.load(pitch_contour_fp)
+            pitch_contour_nonzero = pitch_contour[pitch_contour != 0]
+            if len(pitch_contour_nonzero) > 0:
+                non_zero_pc.append(pitch_contour_nonzero)
+
+        if len(non_zero_pc) > 0:
+            non_zero_pc = torch.cat(non_zero_pc)
+            pitch_mean = non_zero_pc.mean().item()
+            pitch_std = non_zero_pc.std().item()
+            valid = True
+
+            if not _is_valid_pitch(pitch_mean, pitch_std):
+                print("invalid pitch: {}".format(speaker))
+                pitch_mean = 212.0
+                pitch_std = 70.0
+                valid = "False"
+        else:
+            print("could not find pitch contour for speaker {}".format(speaker))
+            valid = "False"
+            pitch_mean = 212.0
+            pitch_std = 70.0
+
+        speaker_pitch_stats[speaker] = {"pitch_mean": pitch_mean, "pitch_std": pitch_std, "valid": valid}
+
+    with open(os.path.join(sup_data_dir, "speaker_pitch_stats.json"), "w") as f:
+        print(sup_data_dir)
+        print("speaker_pitch_stats.json")
+        json.dump(speaker_pitch_stats, f)
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Evaluate the model')
+    parser.add_argument(
+        '--ssl_model_ckpt_path',
+        type=str,
+        default="/home/pneekhara/NeMo2022/tensorboards/ConformerModels/ConformerCompatibleTry3/ConformerCompatible_Epoch3.ckpt",
+    )
+    parser.add_argument('--manifest_path', type=str, default="/home/pneekhara/NeMo2022/libri_val_formatted.json")
+    parser.add_argument('--batch_size', type=int, default=32)
+    parser.add_argument('--ssl_content_emb_type', type=str, default="embedding_and_probs")
+    parser.add_argument('--use_unique_tokens', type=int, default=1)
+    parser.add_argument('--num_workers', type=int, default=8)
+    parser.add_argument('--pool_workers', type=int, default=30)
+    parser.add_argument('--compute_pitch_contours', type=int, default=1)
+    parser.add_argument('--num_pitch_per_speaker', type=int, default=None)  # saves time.
+
+    args = parser.parse_args()
+
+    device = "cuda"
```

Comment:
May want to check whether GPU is available

Reply:
done